components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/utils/mappings/APIMappingUtil.java

```diff
@@ -228,6 +227,67 @@ public static APIDTO fromAPItoDTO(org.wso2.carbon.apimgt.api.model.API model) th
 
     }
 
+    private static String mapVisibilityFromDTOtoAPI(APIDTO.VisibilityEnum visibility) {
+        switch (visibility) {
+            case PUBLIC:
+                return APIConstants.API_GLOBAL_VISIBILITY;
+            case PRIVATE:
+                return APIConstants.API_PRIVATE_VISIBILITY;
+            case RESTRICTED:
+                return APIConstants.API_RESTRICTED_VISIBILITY;
+            case CONTROLLED:
+                return APIConstants.API_CONTROLLED_VISIBILITY;
+            default:
+                return null; // how to handle this?
+        }
+    }
+    private static APIDTO.VisibilityEnum mapVisibilityFromAPItoDTO(String visibility) {
+        switch (visibility) { //public, private,controlled, restricted
+            case APIConstants.API_GLOBAL_VISIBILITY :
+                return APIDTO.VisibilityEnum.PUBLIC;
+            case APIConstants.API_PRIVATE_VISIBILITY :
+                return APIDTO.VisibilityEnum.PRIVATE;
+            case APIConstants.API_RESTRICTED_VISIBILITY :
+                return APIDTO.VisibilityEnum.RESTRICTED;
+            case APIConstants.API_CONTROLLED_VISIBILITY :
+                return APIDTO.VisibilityEnum.CONTROLLED;
+            default:
+                return null; // how to handle this?
+        }
+    }
+
+    private static APIDTO.SubscriptionAvailabilityEnum mapSubscriptionAvailabilityFromAPItoDTO(
+            String subscriptionAvailability) {
+
+        switch (subscriptionAvailability) {
+            case APIConstants.SUBSCRIPTION_TO_CURRENT_TENANT :
+                return APIDTO.SubscriptionAvailabilityEnum.current_tenant;
+            case APIConstants.SUBSCRIPTION_TO_ALL_TENANTS :
+                return APIDTO.SubscriptionAvailabilityEnum.all_tenants;
+            case APIConstants.SUBSCRIPTION_TO_SPECIFIC_TENANTS :
+                return APIDTO.SubscriptionAvailabilityEnum.specific_tenants;
+            default:
+                return null; // how to handle this?
+        }
+
+    }
+
+    private static String mapSubscriptionAvailabilityFromDTOtoAPI(
```

Comment:
Please consider that one method is mapping DTO to backend model while the other is mapping vice versa.
Method comments were added in PR #1091